### ISSUE TITLE
Week 08  - Day 4  - Finish the Safari

### ISSUE THREAD

**Issue by mdewey** [GarrettAnderson/assignments#30]
# Finish the Safari

For this assignment, you should take your safari API that was already created, and create a simple react app that consumes that API.

## Objectives

- Work with an API that you created

## Requirements

- Add Endpoints with more verbs
- Create a React app that consumes an API that you created

### Explorer Mode (C#)

- [ ] Add 3 the endpoints to the Safari API
  - [ ] Create a `GET /Animal/{location}` that returns animals of only that location
  - [ ] Create a `PUT /Animal/{id}` endpoint that adds 1 to the count of times seen for that animal (given by id)
  - [ ] Create a `DELETE /Animal/{id}` endpoint that deletes that animal id from the database

- [ ] Create a simple react app, that uses some css make it looks friendly.
- [ ] The react app should:
  - [ ] Display all animals the user has seen
  - [ ] Display all animals seen in the `Jungle`
  - [ ] Remove all animals that I have seen in the `Desert`.
  - [ ] Add all the `CountOfTimesSeen` and get a total number of animals seen
  - [ ] Get the `CountOfTimesSeen` of `lions`, `tigers` and `bears`

Don't be afraid to add endpoints to make your life easier.

### Explorer Mode (Ruby)

*IMPORTANT* To add a react app we need to update our Rails application a little.

#### Setup for adding a React app

- [x] In the `Gemfile`, uncomment the line containing `rack-cors`
- [x] In the terminal, inside your project, run `bundle`
- [x] In the file `cors.rb` change the text `origins 'example.com'` to `origins '*'`
- [x] In the file `cors.rb` uncomment lines 8 through 16. (Line 8 should say `Rails.application.config.middleware.insert_before 0, Rack::Cors do` and line 16 should be an `end`

#### Assignment

- [ ] Add these features to the Safari API
  - [x] Create a `GET /animals?location={location}` that returns animals of only that location
  - [x] Create a `PUT /animals/count/{id}` endpoint that adds 1 to the count of times seen for that animal (given by id)

- [ ] Create a simple react app, that uses some CSS make it looks friendly.
- [ ] The react app should:
  - [x] Display all animals the user has seen
  - [x] Have a button next to each animal to delete it.
  - [x] Below the list of all animals, show the total of `seen_count`
  - [ ] Get the total of `seen_count` for `lions`, `tigers` and `bears`

Don't be afraid to add endpoints to make your life easier.

### Adventure Mode

- [ ] Work on your capstone' front end. This should include, creating the react app, adding react-router and also start thing about the AJAX (axios/fetch) calls your app needs.

**Comment by GarrettAnderson** [comment]
I got caught up with my capstone and didn't plan accordingly in order to finish this assignment. Plan to complete it today.

**Comment by GarrettAnderson** [comment]
https://github.com/GarrettAnderson/Safari_Vacation

**Comment by GarrettAnderson** [comment]
https://github.com/GarrettAnderson/safari2

**Comment by GarrettAnderson** [comment]
https://github.com/GarrettAnderson/safari-2-frontend

**Comment by GarrettAnderson** [comment]
Realized that I did not satisfy the entire assignment bc I was looking at a previous instruction. I added the correct links to my GitHub repos of the backend and frontend of the Safari Adventure assignment. Was able to get the total number of animals to render along with the other requirements. 
* For future update: I think it would be fun to add a button and onClick function on the frontend that will increment the seen_count attribute value.
